src/irrlicht_changes/CGUITTFont.cpp

```diff
@@ -698,6 +691,18 @@ void CGUITTFont::draw(const EnrichedString &text, const core::rect<s32>& positio
 			tmp_source_rects.set_data(&page->render_source_rects[ibegin], i - ibegin);
 			--i;
 
+			if (shadow_offset) {
+				for (size_t i = 0; i < tmp_positions.size(); ++i)
+					tmp_positions[i] += core::vector2di(shadow_offset, shadow_offset);
+
+				u32 new_shadow_alpha = core::clamp(core::round32(shadow_alpha * colprev.getAlpha() / 255.0f), 0, 255);
+				video::SColor shadow_color = video::SColor(new_shadow_alpha, 0, 0, 0);
+				Driver->draw2DImageBatch(page->texture, tmp_positions, tmp_source_rects, clip, shadow_color, true);
+
+				for (size_t i = 0; i < tmp_positions.size(); ++i)
+					tmp_positions[i] -= core::vector2di(shadow_offset, shadow_offset);
+			}
+
 			if (!use_transparency)
 				colprev.color |= 0xff000000;
```

Comment:
Shouldn't this be moved up?

Reply:
technically yes but I'm 99% sure we don't use `use_transparency=false`

Reply:
We don't use it, moved it anyway as I don't want to remove it in this PR.